src/parquet/column_reader.cc

```diff
@@ -243,4 +166,27 @@ bool ColumnReader::ReadNewPage() {
   return true;
 }
 
+ColumnReader* make_column_reader(const parquet::ColumnMetaData* metadata,
```

Comment:
Good point. How about `std::shared_ptr` (which spares us from having to transfer the pointer from `unique_ptr` to `shared_ptr` in `RowGroupReader::Column` (see #23)